File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/ConfigManagerFactory.java
package com.soonyong.mywebcrawler.config.manage;

import android.content.Context;

import java.io.IOException;

public class ConfigManagerFactory {

    private static ConfigManager configManager;

    private ConfigManagerFactory() {

    }

    public static ConfigManager getConfigManager(Context context) throws IOException {
        if (ConfigManagerFactory.configManager == null) {
            ConfigManagerFactory.configManager = new InMemoryConfigManager();
        }
        return ConfigManagerFactory.configManager;
    }
}

File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/ConfigManagerFactory.kt
package com.soonyong.mywebcrawler.config.manage

import android.content.Context
import java.io.IOException

object ConfigManagerFactory {
    private var configManager: ConfigManager? = null
    @Throws(IOException::class)
    fun getConfigManager(context: Context?): ConfigManager? {
        if (configManager == null) {
            configManager = InMemoryConfigManager()
        }
        return configManager
    }
}